master/internal/rm/kubernetesrm/nodes.go

```diff
@@ -1,101 +1,79 @@
 package kubernetesrm
 
 import (
-	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/sirupsen/logrus"
 
 	k8sV1 "k8s.io/api/core/v1"
 	k8Informers "k8s.io/client-go/informers"
 	k8sClient "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 )
 
-// Messages sent by the nodeInformer to itself.
-type (
-	startNodeInformer struct{}
-)
-
-// Messages sent by the nodeInformer to the podsHandler.
-type (
-	nodeStatusUpdate struct {
-		updatedNode *k8sV1.Node
-		deletedNode *k8sV1.Node
-	}
-)
+type nodeCallbackFunc func(*k8sV1.Node, bool)
 
 type nodeInformer struct {
 	informer    k8Informers.SharedInformerFactory
-	podsHandler *actor.Ref
+	nodeHandler nodeCallbackFunc
+	syslog      *logrus.Entry
 	stop        chan struct{}
 }
 
-func newNodeInformer(clientSet k8sClient.Interface, podsHandler *actor.Ref) *nodeInformer {
+func newNodeInformer(clientSet k8sClient.Interface, nodeHandler nodeCallbackFunc) *nodeInformer {
 	return &nodeInformer{
 		informer: k8Informers.NewSharedInformerFactoryWithOptions(
 			clientSet, 0, []k8Informers.SharedInformerOption{}...),
-		podsHandler: podsHandler,
+		nodeHandler: nodeHandler,
+		syslog:      logrus.WithField("component", "nodeInformer"),
 		stop:        make(chan struct{}),
 	}
 }
 
-func (n *nodeInformer) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		ctx.Tell(ctx.Self(), startNodeInformer{})
-
-	case startNodeInformer:
-		if err := n.startNodeInformer(ctx); err != nil {
-			return err
-		}
-
-	case actor.PostStop:
-		ctx.Log().Infof("shutting down node informer")
-		close(n.stop)
-
-	default:
-		ctx.Log().Errorf("unexpected message %T", msg)
-		return actor.ErrUnexpectedMessage(ctx)
-	}
-
-	return nil
-}
-
-func (n *nodeInformer) startNodeInformer(ctx *actor.Context) error {
+func (n *nodeInformer) startNodeInformer() bool {
+	noErrorEncountered := true
 	nodeInformer := n.informer.Core().V1().Nodes().Informer()
 	nodeInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			node, ok := obj.(*k8sV1.Node)
 			if ok {
-				ctx.Log().Debugf("node added %s", node.Name)
-				ctx.Tell(n.podsHandler, nodeStatusUpdate{updatedNode: node})
+				n.syslog.Debugf("node added %s", node.Name)
+				go n.nodeHandler(node, true)
```

Comment:
I added a bool value to `nodeHandler` to distinguish between the updateNode & deleteNode cases, but it doesn't seem easily understandable. Short of adding more comments, or reinstating the nodeStatusUpdate struct, any suggestions to make this more clear?

Reply:
update: instead of the bool, I'm passing `event.WatchType` to the new handler function